contracts/external/policies/ERC20SpendingLimitPolicy.sol

```diff
@@ -167,39 +159,21 @@ contract ERC20SpendingLimitPolicy is IActionPolicy {
      * @dev we do not check if the transfer is from self to self, as this should not be allowed by token itself
      * returns bool => isTransfer, amount spent
      */
-    function _isTokenTransfer(
-        address account,
-        bytes calldata callData
-    )
-        internal
-        pure
-        returns (bool, uint256) 
-    {
+    function _isTokenTransfer(address account, bytes calldata callData) internal pure returns (bool, uint256) {
         bytes4 functionSelector = bytes4(callData[0:4]);
 
-        if (functionSelector == IERC20.approve.selector) {
-            (, uint256 amount) = abi.decode(callData[4:], (address, uint256));
-            // if approve is to account itself, it should revert in the token contract
-            // otherwise it should spend the limit
-            return (true, amount);
-        } else if (functionSelector == bytes4(keccak256("increaseAllowance(address,uint256)"))) {
-            // increase allowance is deprecated interface by OZ, can be used by some tokens
-            (, uint256 amount) = abi.decode(callData[4:], (address, uint256));
-            return (true, amount);
-        } else if (functionSelector == IERC20.transfer.selector) {
+        if (functionSelector == IERC20.transfer.selector) {
             (, uint256 amount) = abi.decode(callData[4:], (address, uint256));
             // if transfer is to account, it should revert in the token contract
             // otherwise it should spend the limit
             return (true, amount);
         } else if (functionSelector == IERC20.transferFrom.selector) {
-            (, address to, uint256 amount) = abi.decode(callData[4:], (address, address, uint256));
-            if(to == account) {
-                // if transfer is from and to account, it should revert in the token contract
-                // if transfer is from somewhere to account, it should not spend the limit, so amount is 0            
+            (address from, address to, uint256 amount) = abi.decode(callData[4:], (address, address, uint256));
+            if (from != account || to == account) {
```

Comment:
If `from != account` that means session involves transferring tokens approved by someone to account, and reducing this approved amount which should be considered spending like spending own tokens.

Reply:
Check this https://github.com/erc7579/smartsessions/pull/122#discussion_r1797720129 convo for reference